src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Parser/ParserUtils.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Text.Json;
+using System.Text;
+using System;
+using Microsoft.Sbom.Parsers.Spdx22SbomParser;
+using Microsoft.Sbom.Exceptions;
+
+namespace Microsoft.Sbom.Parser;
+
+/// <summary>
+/// Utility methods for parsing that are shared by all parsers.
+/// </summary>
+internal class ParserUtils
+{
+    /// <summary>
+    /// Read the next JSON token in the reader from the input buffer.
+    /// If the buffer is small and doesn't contain all the text for the next token, 
+    /// a call to GetMoreBytesFromStream is made to read more data into the buffer.
+    /// </summary>
+    /// <exception cref="EndOfStreamException"></exception>
+    public static void Read(Stream stream, ref byte[] buffer, ref Utf8JsonReader reader)
+    {
+        // If the buffer is empty, refill the buffer.
+        if (buffer[0] == 0)
+        {
+            if (!stream.CanRead || stream.Read(buffer) == 0)
+            {
+                throw new EndOfStreamException();
+            }
+        }
+
+        while (!reader.Read())
+        {
+            // Not enough of the JSON is in the buffer to complete a read.
+            GetMoreBytesFromStream(stream, ref buffer, ref reader);
+        }
+    }
+
+    /// <summary>
+    /// Asserts if the reader is at the current expected token.
+    /// </summary>
+    /// <exception cref="ParserError"></exception>
+    internal static void AssertTokenType(Stream stream, ref Utf8JsonReader reader, JsonTokenType expectedTokenType)
+    {
+        if (reader.TokenType != expectedTokenType)
+        {
+            throw new ParserError($"Expected a '{Constants.JsonTokenStrings[(byte)expectedTokenType]}' at position {stream.Position}");
```

Comment:
>(byte)expectedTokenType [](http://example.com/codeflow?start=76&length=23)

How can you guarantee that Constants.JsonTokenStrings contains the element you are looking for? What is the relationship between JsonTokenType and Constants.JsonTokenStrings?

Reply:
Its an enum, and I have enumerated every element in the enum, so it should always have the required element